### PR TITLE
Enable timespans to be recorded in millis instead of nanos

### PR DESCRIPTION
**Pull request by parkwherever**

### CODE REVIEW

File: stats.go [hunk 1]
```diff
@@ -285,12 +288,20 @@ type timespan struct {
...
 func (ts *timespan) Complete() time.Duration {
 	d := time.Now().Sub(ts.start)
 	ts.timer.time(d)
 	return d
 }
 
+// CompleteAsMS records the elapsed time in ms, and returns the elapsed duration.
+func (ts *timespan) CompleteAsMS() time.Duration {
+	d := time.Now().Sub(ts.start)
+	ts.timer.time(d / time.Millisecond)
```
Comment (charlievieth) [inline stats.go]: `timer.time()` is:
```golang
func (t *timer) time(dur time.Duration) {
	t.sink.FlushTimer(t.name, float64(dur/time.Microsecond))
}
```

So you need to multiply by 1000 or `time.Millisecond / time.Microsecond` not divide.  This currently gives you nanoseconds not milliseconds.

**Comment by charlievieth** [comment]
So I'm torn between the `timer` being consistently wrong (it reports microseconds not milliseconds) or often wrong and occasionally correct.  The part I like about being consistently wrong is that it makes it very easy to figure out what values are being reported and how to fix them.  It also eliminates the chance of a metric being reported two ways, one call using `Complete()` the other `CompleteAsMS()`.  Additionally, if at some point a metric was moved from micro `Complete()` to milliseconds `CompleteAsMS()` it would make analyzing historical data very difficult.

While annoying, I do believe that its best that we stay consistent in the timer metric we report.

File: stats.go (hunk 1 above)
Comment (parkwherever) [reply stats.go]: doesn't this currently give micros / 10e6 then, so seconds as written?

Comment (charlievieth) [reply stats.go]: Yep, this gives seconds.  But the method name and comment indicate that this records milliseconds.

**Pull request closed without merging**